src/transaction/hooks/useTransactionType.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMemo } from 'react';
+import type { ContractFunctionParameters } from 'viem';
+import type { WalletCapabilities } from '../../types';
+import {
+  TRANSACTION_TYPE_CALLS,
+  TRANSACTION_TYPE_CONTRACTS,
+} from '../constants';
+import type { Call } from '../types';
+
+export const useTransactionType = ({
+  calls,
+  contracts,
+  transactionStatuses,
+  walletCapabilities,
+}: {
+  calls?: Call[];
+  contracts?: ContractFunctionParameters[];
+  transactionStatuses: {
+    [TRANSACTION_TYPE_CALLS]: {
+      single: string;
+      batch: string;
+    };
+    [TRANSACTION_TYPE_CONTRACTS]: {
+      single: string;
+      batch: string;
+    };
+  };
+  walletCapabilities: WalletCapabilities;
+}) => {
+  return useMemo(() => {
+    const transactionType =
+      calls || !contracts ? TRANSACTION_TYPE_CALLS : TRANSACTION_TYPE_CONTRACTS;
+
+    const singleOrBatched = walletCapabilities.hasAtomicBatch
+      ? 'batch'
+      : 'single';
```

Comment:
Just a thought, we could also make this `isBatched` and set it to a boolean

Reply:
oh yeah, boolean always with `is` or `has` prefix

Reply:
In this case because `singleOrBatched` is a `string`, we probably want to keep it as "noun" to use inside the `transactionStatuses` as key map.